evalml/pipelines/components/estimators/classifiers/lightgbm_classifier.py

```diff
@@ -74,10 +74,14 @@ def __init__(self, boosting_type="gbdt", learning_rate=0.1, n_estimators=100, ma
 
     def _encode_categories(self, X, fit=False):
         """Encodes each categorical feature using ordinal encoding."""
-        X_encoded = _convert_to_woodwork_structure(X)
-        X_encoded = _rename_column_names_to_numeric(X_encoded)
-        cat_cols = list(X_encoded.select('category').columns)
-        X_encoded = _convert_woodwork_types_wrapper(X_encoded.to_dataframe())
+        X = _convert_to_woodwork_structure(X)
+        cat_cols = list(X.select('category').columns)
+        X = _convert_woodwork_types_wrapper(X.to_dataframe())
+
+        X_encoded = _rename_column_names_to_numeric(X)
+        rename_cols_dict = dict(zip(X.columns, X_encoded.columns))
+        cat_cols = [rename_cols_dict[col] for col in cat_cols]
```

Comment:
Since we moved this to after conversion to dataframe (need dataframe to set column names easily for tuples), need to find mapping from old to new names and update.